Warnings in various unit tests

Working on new unit tests, I ran the existing tests locally and noticed various warnings and a skipped test. I did not yet investigate any of this. For now, this is just a collection of the warnings, so we can see if they need further attention.

Running `nosetests3 skrf -c nose.cfg`, the following 8 warnings were reported:

1. 
```
..................................................................................../usr/local/lib/python3.8/dist-packages/numpy/core/_asarray.py:102: VisibleDeprecationWarning: Creating an ndarray from nested sequences exceeding the maximum number of dimensions of 32 is deprecated. If you mean to do this, you must specify 'dtype=object' when creating the ndarray.
  return array(a, dtype, copy=False, order=order)
.............................S...../usr/lib/python3.8/unittest/case.py:608: RuntimeWarning: TestResult has no addExpectedFailure method, reporting as passes
  warnings.warn("TestResult has no addExpectedFailure method, reporting as passes",
....../home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6213: RuntimeWarning: divide by zero encountered in true_divide
  s_act[fidx] = npy.matmul(s[fidx], a) / a
...F....................................test_create_linear_sweep (test_frequency.FrequencyTestCase) ... ok
```

2.
```
................/home/vinc/PycharmProjects/scikit-rf/skrf/mathFunctions.py:162: RuntimeWarning: divide by zero encountered in log10
  out = 20 * npy.log10(input)
/home/vinc/PycharmProjects/scikit-rf/skrf/mathFunctions.py:177: RuntimeWarning: invalid value encountered in log10
  out = 10 * npy.log10(input)
../home/vinc/PycharmProjects/scikit-rf/skrf/mathFunctions.py:162: RuntimeWarning: invalid value encountered in log10
  out = 20 * npy.log10(input)
....test_add (test_network.NetworkTestCase) ... ok
```

3.
`test_connect_fast (test_network.NetworkTestCase) ... skipped 'not supporting this function currently '`

4.
```
test_connect_nport_2port (test_network.NetworkTestCase) ... /home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6117: RuntimeWarning: divide by zero encountered in true_divide
  result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6117: RuntimeWarning: invalid value encountered in true_divide
  result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6117: RuntimeWarning: invalid value encountered in multiply
  result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:4590: RuntimeWarning: invalid value encountered in true_divide
  (A[:, k, j] * A[:, i, l] * (1 - A[:, l, k]) + \
```

5.
```
Test the active s-parameters of a 2-ports network ... /home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6213: RuntimeWarning: divide by zero encountered in true_divide
  s_act[fidx] = npy.matmul(s[fidx], a) / a
```

6.
```
Test the active vswr-parameters of a 2-ports network ... /home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6213: RuntimeWarning: divide by zero encountered in true_divide
  s_act[fidx] = npy.matmul(s[fidx], a) / a
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6343: RuntimeWarning: invalid value encountered in true_divide
  vswr_act[fidx] = (1 + npy.abs(s_act[fidx]))/(1 - npy.abs(s_act[fidx]))
```

7.
```
...S./home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6117: RuntimeWarning: divide by zero encountered in true_divide
  result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6117: RuntimeWarning: invalid value encountered in true_divide
  result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6117: RuntimeWarning: invalid value encountered in multiply
  result[:, 0, 1] = (1 - gamma) * npy.sqrt(1.0 * z2 / z1)
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:4590: RuntimeWarning: invalid value encountered in true_divide
  (A[:, k, j] * A[:, i, l] * (1 - A[:, l, k]) + \
.................................../home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6213: RuntimeWarning: divide by zero encountered in true_divide
  s_act[fidx] = npy.matmul(s[fidx], a) / a
............./home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6213: RuntimeWarning: divide by zero encountered in true_divide
  s_act[fidx] = npy.matmul(s[fidx], a) / a
/home/vinc/PycharmProjects/scikit-rf/skrf/network.py:6343: RuntimeWarning: invalid value encountered in true_divide
  vswr_act[fidx] = (1 + npy.abs(s_act[fidx]))/(1 - npy.abs(s_act[fidx]))
.......................test_constructor (test_networkSet.NetworkSetTestCase)
```

8.
```
FAIL: The networkx package should be available to run these tests
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vinc/PycharmProjects/scikit-rf/skrf/tests/test_circuit.py", line 737, in test_is_networkx_available
    self.failUnless('networkx' in sys.modules)
AssertionError: False is not true
```
 I have networkx installed. Could it be misconfigured?

--- a/doc/source/tutorials/Networks.ipynb
+++ b/doc/source/tutorials/Networks.ipynb
@@ -801,7 +801,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "line.interpolate_from_f(line1.frequency)"
+    "line.interpolate(line1.frequency)"
    ]
   },
   {
@@ -951,7 +951,7 @@
   "anaconda-cloud": {},
   "celltoolbar": "Raw Cell Format",
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 3 (ipykernel)",
    "language": "python",
    "name": "python3"
   },
@@ -965,7 +965,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.8"
+   "version": "3.9.7"
   }
  },
  "nbformat": 4,

--- a/skrf/network.py
+++ b/skrf/network.py
@@ -2422,7 +2422,7 @@ class Network(object):
 
         The input 'freq_or_n` can be either a new
         :class:`~skrf.frequency.Frequency` or an `int`, or a new
-        frequency vector (in hz).
+        frequency vector (in Hz).
 
         This interpolates  a given `basis`, ie s, z, y, etc, in the
         coordinate system defined by `coord` like polar or cartesian.
@@ -2572,68 +2572,35 @@ class Network(object):
               result.noise_freq = new_frequency
         return result
 
-    def interpolate_self_npoints(self, npoints: int, **kwargs) -> None:
-        """
-
-        Interpolate network based on a new number of frequency points
-
-
-        Note
-        ----
-        The function :func:`~Network.resample` is an alias for
-        :func:`~Network.interpolate_self_npoints`.
-
-        Parameters
-        ----------
-        npoints : int
-                number of frequency points
-        **kwargs : keyword arguments
-                passed to :func:`scipy.interpolate.interp1d` initializer.
-
-        See Also
-        --------
-        interpolate_self : same functionality but takes a Frequency
-                object
-        interpolate : same functionality but takes a Frequency
-                object and returns a new Network, instead of updating
-                itself.
-
-
-        Examples
-        --------
-        .. ipython::
-
-            @suppress
-            In [21]: import skrf as rf
-
-            In [21]: n = rf.data.ring_slot
-
-            In [21]: n
-
-            In [21]: n.resample(501) # resample is an alias
-
-            In [21]: n
-
-        """
-        warnings.warn('Use interpolate_self', DeprecationWarning)
-        new_frequency = self.frequency.copy()
-        new_frequency.npoints = npoints
-        self.interpolate_self(new_frequency, **kwargs)
-
     def interpolate_self(self, freq_or_n: Union[Frequency, NumberLike], **kwargs) -> None:
         """
-        Interpolates s-parameters given a new
+        Interpolate the current Network along frequency axis (inplace).
 
-        :class:'~skrf.frequency.Frequency' object.
+        The input 'freq_or_n` can be either a new
+        :class:`~skrf.frequency.Frequency` or an `int`, or a new
+        frequency vector (in Hz).
 
         See :func:`~Network.interpolate` for more information.
 
         Parameters
         ----------
-        new_frequency : :class:`~skrf.frequency.Frequency`
-                frequency information to interpolate at
+        freq_or_n : :class:`~skrf.frequency.Frequency` or int or list-like
+            The new frequency over which to interpolate. this arg may be
+            one of the following:
+
+            * a new :class:`~skrf.frequency.Frequency` object
+
+            * an int: the current frequency span is resampled linearly.
+
+            * a list-like: create a new frequency using :meth:`~skrf.frequency.Frequency.from_f`
+
         **kwargs : keyword arguments
                 passed to :func:`scipy.interpolate.interp1d` initializer.
+
+        Returns
+        -------
+        None
+            The interpolation is performed inplace.
 
         See Also
         --------

--- a/skrf/network.py
+++ b/skrf/network.py
@@ -2616,46 +2616,6 @@ class Network(object):
     ##convenience
     resample = interpolate_self
 
-    def interpolate_from_f(self, f: Frequency, interp_kwargs: dict = {}, **kwargs) -> 'Network':
-        r"""
-        Interpolates s-parameters from a frequency vector.
-
-        Given a frequency vector, and optionally a `unit` (see \*\*kwargs)
-        , interpolate the networks s-parameters linearly in real and
-        imaginary components.
-
-        See :func:`~Network.interpolate` for more information.
-
-
-        Note
-        ----
-        This creates a new :class:`~skrf.frequency.Frequency`, object
-        using the method :func:`~skrf.frequency.Frequency.from_f`,
-        and then calls :func:`~Network.interpolate_self`.
-
-
-        Parameters
-        ----------
-        new_frequency : :class:`~skrf.frequency.Frequency`
-            frequency information to interpolate at
-        interp_kwargs :
-            dictionary of kwargs to be passed through to
-            :func:`scipy.interpolate.interpolate.interp1d`
-        \*\*kwargs :
-            passed to :func:`scipy.interpolate.interp1d` initializer.
-
-        See Also
-        --------
-        resample
-        interpolate
-        interpolate_self
-        """
-        warnings.warn('Use interpolate', DeprecationWarning)
-        return self.interpolate(freq_or_n=f, f_kwargs=kwargs,
-                                **interp_kwargs)
-        # freq = Frequency.from_f(f,**kwargs)
-        # self.interpolate_self(freq, **interp_kwargs)
-
     def extrapolate_to_dc(self, points: int = None, dc_sparam: NumberLike = None,
                           kind: str = 'cubic', coords: str = 'cart',
                           **kwargs) -> 'Network':

--- a/skrf/network.py
+++ b/skrf/network.py
@@ -2690,7 +2690,7 @@ class Network(object):
         # self.interpolate_self(freq, **interp_kwargs)
 
     def extrapolate_to_dc(self, points: int = None, dc_sparam: NumberLike = None,
-                          kind: str = 'warn', coords: str = 'cart',
+                          kind: str = 'cubic', coords: str = 'cart',
                           **kwargs) -> 'Network':
         """
         Extrapolate S-parameters down to 0 Hz and interpolate to uniform spacing.
@@ -2710,8 +2710,7 @@ class Network(object):
         dc_sparam : class:`npy.ndarray` or None
             NxN S-parameters matrix at 0 Hz.
             If None S-parameters at 0 Hz are determined by linear extrapolation.
-        kind : str or int, default is 'rational'
-            Default value will be changed to 'cubic' in future version.
+        kind : str or int, default is 'cubic'
             Specifies the kind of interpolation as a string ('linear',
             'nearest', 'zero', 'slinear', 'quadratic, 'cubic') or
             as an integer specifying the order of the spline
@@ -2737,14 +2736,6 @@ class Network(object):
         impulse_response
         step_response
         """
-        if kind == 'warn':
-            warnings.warn("The default `kind` parameter will change "
-                          "from `rational` to `cubic` in future versions."
-                          "Use `kind='rational'` to keep the old default."
-                          "To silent this warning, explitly define `kind`.",
-                          category=DeprecationWarning, stacklevel=2)
-            kind = 'rational'
-
         result = self.copy()
 
         if self.frequency.f[0] == 0:

--- a/skrf/network.py
+++ b/skrf/network.py
@@ -5473,8 +5473,7 @@ def z2y(z: npy.ndarray) -> npy.ndarray:
     .. [#] http://en.wikipedia.org/wiki/impedance_parameters
     .. [#] http://en.wikipedia.org/wiki/Admittance_parameters
     """
-    return npy.array([npy.mat(z[f, :, :]) ** -1 for f in range(z.shape[0])])
-
+    return npy.linalg.inv(z)
 
 def z2t(z: npy.ndarray) -> NoReturn:
     """
@@ -5860,7 +5859,7 @@ def y2z(y: npy.ndarray) -> npy.ndarray:
     .. [#] http://en.wikipedia.org/wiki/Admittance_parameters
     .. [#] http://en.wikipedia.org/wiki/impedance_parameters
     """
-    return npy.array([npy.mat(y[f, :, :]) ** -1 for f in range(y.shape[0])])
+    return npy.linalg.inv(y)
 
 
 def y2t(y: npy.ndarray) -> NoReturn:

--- a/skrf/network.py
+++ b/skrf/network.py
@@ -6668,13 +6668,9 @@ def s2s_active(s: npy.ndarray, a:npy.ndarray) -> npy.ndarray:
     .. [#] D. Williams, IEEE Microw. Mag. 14, 38 (2013).
 
     """
-    # TODO : vectorize the for loop
-    nfreqs, nports, nports = s.shape
-    s_act = npy.zeros((nfreqs, nports), dtype='complex')
-    s[ s == 0 ] = 1e-12  # solve numerical singularity
-
-    for fidx in range(s.shape[0]):
-        s_act[fidx] = npy.matmul(s[fidx], a) / a
+    a = npy.asarray(a, dtype=complex)
+    a[a == 0] = 1e-12  # solve numerical singularity
+    s_act = npy.einsum('fmi,i,m->fm', s, a, npy.reciprocal(a))
     return s_act  # shape : (n_freqs, n_ports)
 
 def s2z_active(s: npy.ndarray, z0: NumberLike, a: npy.ndarray) -> npy.ndarray:
@@ -6716,11 +6712,8 @@ def s2z_active(s: npy.ndarray, z0: NumberLike, a: npy.ndarray) -> npy.ndarray:
     # TODO : vectorize the for loop
     nfreqs, nports, nports = s.shape
     z0 = fix_z0_shape(z0, nfreqs, nports)
-    z_act = npy.zeros((nfreqs, nports), dtype='complex')
     s_act = s2s_active(s, a)
-
-    for fidx in range(s.shape[0]):
-        z_act[fidx] = z0[fidx] * (1 + s_act[fidx])/(1 - s_act[fidx])
+    z_act = npy.einsum('fp,fp,fp->fp', z0, 1 + s_act, npy.reciprocal(1 - s_act))
     return z_act
 
 def s2y_active(s: npy.ndarray, z0: NumberLike, a: npy.ndarray) -> npy.ndarray:
@@ -6760,11 +6753,8 @@ def s2y_active(s: npy.ndarray, z0: NumberLike, a: npy.ndarray) -> npy.ndarray:
     """
     nfreqs, nports, nports = s.shape
     z0 = fix_z0_shape(z0, nfreqs, nports)
-    y_act = npy.zeros((nfreqs, nports), dtype='complex')
     s_act = s2s_active(s, a)
-
-    for fidx in range(s.shape[0]):
-        y_act[fidx] = 1/z0[fidx] * (1 - s_act[fidx])/(1 + s_act[fidx])
+    y_act = npy.einsum('fp,fp,fp->fp', npy.reciprocal(z0), 1 - s_act, npy.reciprocal(1 + s_act))        
     return y_act
 
 def s2vswr_active(s: npy.ndarray, a: npy.ndarray) -> npy.ndarray:
@@ -6799,13 +6789,8 @@ def s2vswr_active(s: npy.ndarray, a: npy.ndarray) -> npy.ndarray:
         s2z_active : active Z-parameters
         s2y_active : active Y-parameters
     """
-    nfreqs, nports, nports = s.shape
-    vswr_act = npy.zeros((nfreqs, nports), dtype='complex')
     s_act = s2s_active(s, a)
-
-    for fidx in range(s.shape[0]):
-        vswr_act[fidx] = (1 + npy.abs(s_act[fidx]))/(1 - npy.abs(s_act[fidx]))
-
+    vswr_act = npy.einsum('fp,fp->fp', (1 + npy.abs(s_act)), npy.reciprocal(1 - npy.abs(s_act)))
     return vswr_act
 
 

--- a/skrf/tests/test_network.py
+++ b/skrf/tests/test_network.py
@@ -680,9 +680,9 @@ class NetworkTestCase(unittest.TestCase):
         self.assertTrue(b.z0[1] == 0.5*(a.z0[0] + a.z0[1]))
         self.assertTrue(b.z0[-1] == a.z0[-1])
 
-    def test_interpolate_self_npoints(self):
+    def test_interpolate_self(self):
         a = rf.N(f=[1,2],s=[1+2j, 3+4j],z0=1)
-        a.interpolate_self_npoints(4)
+        a.interpolate_self(4)
         # TODO: numerically test for correct interpolation
 
     def test_interpolate_from_f(self):

--- a/skrf/tests/test_network.py
+++ b/skrf/tests/test_network.py
@@ -681,13 +681,8 @@ class NetworkTestCase(unittest.TestCase):
         self.assertTrue(b.z0[-1] == a.z0[-1])
 
     def test_interpolate_self(self):
-        a = rf.N(f=[1,2],s=[1+2j, 3+4j],z0=1)
+        a = rf.N(f=[1,2], s=[1+2j, 3+4j], z0=1)
         a.interpolate_self(4)
-        # TODO: numerically test for correct interpolation
-
-    def test_interpolate_from_f(self):
-        a = rf.N(f=[1,2],s=[1+2j, 3+4j],z0=1)
-        a.interpolate_from_f(npy.linspace(1,2,4), unit='ghz')
         # TODO: numerically test for correct interpolation
 
     def test_slicer(self):

--- a/skrf/tests/test_network.py
+++ b/skrf/tests/test_network.py
@@ -251,7 +251,7 @@ class NetworkTestCase(unittest.TestCase):
             nport = rf.Network()
             nport.frequency = freq
             nport.s = npy.zeros((10, nport_portnum, nport_portnum))
-            nport.z0 = npy.arange(nport_portnum)
+            nport.z0 = npy.arange(start=1, stop=nport_portnum + 1)
 
             # Connect the line to each port and check for port impedance
             for port in range(nport_portnum):

